packages/backend/src/telemetry/transaction-amount.ts

```diff
@@ -0,0 +1,53 @@
+import { ValueType } from '@opentelemetry/api'
+import { ConvertError, RatesService } from '../rates/service'
+import { ConvertOptions } from '../rates/util'
+import { TelemetryService } from './service'
+import { privacy } from './privacy'
+import { Asset } from '../rates/util'
+
+export async function collectTelemetryAmount(
+  telemetryService: TelemetryService,
+  aseRates: RatesService,
+  { amount, asset }: { amount: bigint; asset: Asset }
+) {
+  if (!amount) {
+    return
+  }
+  const convertOptions = {
+    sourceAmount: amount,
+    sourceAsset: { code: asset.code, scale: asset.scale },
+    destinationAsset: {
+      code: telemetryService.getBaseAssetCode(),
+      scale: telemetryService.getBaseScale()
+    }
+  }
+  const converted = await convertAmount(
+    aseRates,
+    telemetryService.getRatesService(),
+    convertOptions
+  )
+  if (converted === ConvertError.InvalidDestinationPrice) {
+    return
+  }
+
+  telemetryService
+    .getOrCreate('transactions_amount', {
+      description: 'Amount sent through the network',
+      valueType: ValueType.DOUBLE
+    })
+    .add(privacy.applyPrivacy(Number(converted)))
```

Comment:
can `.add` throw? we should make sure to handle this to not break the main process. It's ok if telemetry doesn't end up sending a metric, but wouldn't be nice obviously if it prevents a trx from completing

Reply:
Caught and logged all bussines code here to make sure that does not happen. Have another 👀